src/utils/TagUtils.ts

```diff
@@ -83,4 +83,9 @@ export default class TagUtils {
 
         return tagValue;
     }
+
+    static limitCategoriesToMaxCount(categories: Categories, max: number): Categories {
```

Comment:
```
static limitCategoriesToMaxCount(categories: Categories, max: number): Categories {
        categories.tags = categories.tags.slice(0, max);
        return categories;
    }
```
This method modifies the original object while pretends it returns a copy.
Can we actually make it not changing the underlying object? Example:
```
static limitCategoriesToMaxCount(categories: Categories, max: number): Categories {
        const limitedCategories = {
             ...categories,
             tags: categories.tags.slice(0, max),
        };
        return limitedCategories;
    }
```

Reply:
Done.